hashing/src/chunk_with_proof.rs

```diff
@@ -105,6 +128,29 @@ mod test {
 
     use crate::chunk_with_proof::ChunkWithProof;
 
+    fn prepare_bytes(length: usize) -> Vec<u8> {
+        let mut rng = rand::thread_rng();
+
+        let mut v = Vec::with_capacity(length);
+        (0..length).into_iter().for_each(|_| v.push(rng.gen()));
+        v
```

Comment:
Maybe a bit simpler:

```suggestion
        (0..length).into_iter().map(|_| rng.gen()).collect()
```

Reply:
It was implemented using the `with_capacity()` approach, because the data could potentially be quite big.

Unless `collect()` can somehow infer the total number of elements, but I don't believe so.

Reply:
Ok, changing the code to more concise one.